src/main/java/com/es/phoneshop/model/cart/HttpSessionCartService.java

```diff
@@ -0,0 +1,98 @@
+package com.es.phoneshop.model.cart;
+
+import com.es.phoneshop.model.exception.IllegalQuantityException;
+import com.es.phoneshop.model.exception.LackOfStockException;
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.model.product.ProductDao;
+import javax.servlet.http.HttpServletRequest;
+import java.math.BigDecimal;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.Locale;
+import java.util.Optional;
+
+public class HttpSessionCartService implements CartService{
+    private static final CartService INSTANCE = new HttpSessionCartService();
+    private static final String CART_ATTRIBUTE = "cart";
+    private static final String SUCCESS_PARSING = "success parsing";
+    private ProductDao productDao = ArrayListProductDao.getInstance();
+
+    private HttpSessionCartService() {
+    }
+
+    public static CartService getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public String add(Cart cart, Long productId, String quantity, Locale locale) throws LackOfStockException,
+            IllegalQuantityException, NumberFormatException {
+        Product product = productDao.getProduct(productId);
+        String resultOfParsingQuantity = parseQuantity(locale, quantity, product.getStock());
+        if (resultOfParsingQuantity == SUCCESS_PARSING) {
+            int quantityInt = Integer.parseInt(quantity);
```

Comment:
Can we avoid some sort of double parsing in parseQuantity() and Integer.parseInt() ? can we replace this with single parsing in parseQuantity() method?